backend/api/views/games.py

```diff
@@ -246,18 +261,65 @@ def get_giantbomb_data(game_name, game_system):
     gb_dict["thumbnail"] = ""
     gb_dict["image"] = ""
 
-    gb_data = json.loads(
-        requests.get(url=gb_url, params=gb_params, headers=headers).json()
-    )
-    if len(gb_data["results"]) == 0:
-        return gb_dict
-    for result in gb_data["results"]:
-        if game_name.lower() == result["name"].lower():
-            if result["deck"] is not None:
-                gb_dict["description"] = result["deck"]
-            if result["image"]["icon_url"] is not None:
-                gb_dict["thumbnail"] = result["image"]["icon_url"]
-            if result["image"]["small_url"] is not None:
-                gb_dict["image"] = result["image"]["small_url"]
-            return gb_dict
+    gb_data = requests.get(url=gb_url, params=gb_params, headers=headers).json()
+
+    if len(gb_data["results"]) != 0:
+        best_match = gb_data["results"][0]
+        best_similarity = 0
+        # look for closest match
+        for result in gb_data["results"]:
+            similarity = SequenceMatcher(
+                None, game_name.lower(), result["name"].lower()
+            ).ratio()
+            if similarity > best_similarity:
+                best_match = result
+                best_similarity = similarity
+        if best_match["deck"] is not None:
+            gb_dict["description"] = best_match["deck"]
+        if best_match["image"]["icon_url"] is not None:
+            gb_dict["thumbnail"] = best_match["image"]["icon_url"]
+        if best_match["image"]["small_url"] is not None:
+            gb_dict["image"] = best_match["image"]["small_url"]
+
+    while (
+        gb_dict["description"] == ""
+        and gb_dict["image"] == ""
+        and gb_dict["thumbnail"] == ""
+        and modified_name != ""
+    ):
+        # remove last word and retry
```

Comment:
Maybe just have a if statement that removes the last word if it's not the first time through the while loop?
Just to try and remove the repetitive code above the while loop.
Something like
boolean remove_end = false
while {
      if (remove_end) : 
             remove the last word
      get results from api
      look for closest match
      remove_end = true
}